# Question 4 hypothesis test

You are provided with a data set called __iData.csv__ containing information on births recorded in a particular state. Here is a dictionary on the variables included:

fage: father’s age in years. 

mage: mother’s age in years.

mature: maturity status of mother.

weeks: length of pregnancy in weeks.

premie: whether the birth was classified as premature (premie) or full-term.

visits: number of hospital visits during pregnancy.

marital: whether mother is married or not married at birth.

gained: weight gained by mother during pregnancy in pounds.

weight: weight of the baby at birth in pounds.

lowbirthweight: whether baby was classified as low birthweight (low) or not (not low).

gender: gender of the baby, female or male.

habit: status of the mother as a nonsmoker or a smoker.

whitemom: whether mom is white or not white.


__Total marks 15__. Each sub-question a, b, c worths 5 marks.

In [1]:
library(dplyr)
library(e1071)


Attaching package: ‘dplyr’

The following objects are masked from ‘package:stats’:

    filter, lag

The following objects are masked from ‘package:base’:

    intersect, setdiff, setequal, union



a) It is claimed that mothers' smoking habit does not have any effect on the birth weight of the babies. To verity this claim, you need to complete the following tasks: 
a.1. Construct a significance test, by writing a null hypothesis, and an alternative hypothesis,

Since the claim is that mothers' smoking habit does <font color='blue'> not</font> have any effect on the birth weight of the babies and we want to reject our hypothesis, we set $H0$ sentence structure to negates the claim

$H0$: smoking habit <font color='red'>has</font> impact on the birth weight of the babies.

$H1$: smoking habit does not have impact on the birth weight of the babies.

### a.2. perform the test using an appropriate R function,

In [26]:
# Note: Based on Kelvin's comment on Moodle, I deleted the last 3 rows before reading the data into Jupyter

MyData <- read.table(file="iData.csv", header=TRUE, fill=TRUE,sep=",")

# selecting the habit and weigth columns only
smoking <- MyData %>%
  select("habit","weight")

# selecting rows which habit is equal to nonsmoker
nonsmoker <- smoking[smoking$habit == "nonsmoker",]

# selecting rows which habit is equal to smoker
smoker <- smoking[smoking$habit == "smoker",]

# Printing results:
cat("There are ", 
    nrow(nonsmoker), 
   " nonsmoker moms and ",
   nrow(smoker),
   " smoker moms, which makes to ",
    nrow(nonsmoker) + nrow(smoker),
    "rows of data, based on the dataset.\n\n")
cat("The average weight of babies born from nonsmoker mom is equal to: --> ", mean(nonsmoker$weight),"\n\n")
cat("The average weight of babies born from smoker mom is equal to: --> ", mean(smoker$weight),"\n\n")

There are  716  nonsmoker moms and  84  smoker moms, which makes to  800 rows of data, based on the dataset.

The average weight of babies born from nonsmoker mom is equal to: -->  7.24676 

The average weight of babies born from smoker mom is equal to: -->  6.886429 



In [16]:
# I used Welch independent 2-group t-test

# This is testing if the two means are equal.

# If they are not equal --> Our cannot reject that smoking has impact on babies weight

t.test(nonsmoker$weight,smoker$weigh)


	Welch Two Sample t-test

data:  nonsmoker$weight and smoker$weigh
t = 2.3625, df = 108.54, p-value = 0.01994
alternative hypothesis: true difference in means is not equal to 0
95 percent confidence interval:
 0.05801978 0.66264264
sample estimates:
mean of x mean of y 
 7.246760  6.886429 


### a.3. Make a conclusion and explain your findings.

#### p-value = 0.01994 --> result of t-test is that the two means are significantly different

Considering that P-value is way less than 0.05, we conclude that the average weight of children whose mom are smoker <font color='blue'>significantly differs</font> (lower) from the average of  children whose mom are not smoker.

#### Considering that:
There are  716  nonsmoker moms and  84  smoker moms, which makes to  800 rows of data, based on the dataset.

The average weight of babies born from nonsmoker mom is equal to: -->  7.24676 

The average weight of babies born from smoker mom is equal to: -->  6.886429

# <font color='orange'>Findings</font>
As a result, we <font color='orange'>cannot reject</font> $H0$. We also do not accept $H1$ which was the claim mentioned in part a of the question.

Furthermore, since my code (t.test(nonsmoker$weight,smoker$weigh)) is testing if the two means are equal assuming that they are equal, and in results we see "alternative hypothesis: true difference in means is not equal to 0", we understand that they are not same and significantly different and again we cannot reject H0.

We witness that data of non-smoker moms are more accurate (because of having higher number of samples) than those of smoker moms (716 vs 84). I argue that we have to gather further data on the babies with smoker moms and run the analysis further to find more reliable interpretations.



## b) Calculate a 99% confidence interval for the average length of pregnancies.

In [27]:
# At the first step, I need to find the following variables

# xmean <- mean(MyData$weeks)
cat("Xbar is equal to : ",xmean, " \n\nThere are ",nrow(MyData)," number of rows in the data \n\n")

MyData$weekspower2 <- MyData$weeks^2

#sum(x^2)
sumpower <- sum(MyData$weekspower2)

#times(xmean)^2
xmeantimes <- (nrow(MyData))*(xmean^2)

s2 <- (sumpower-xmeantimes)/(nrow(MyData)-1)
s <- sqrt(s2)
cat("Standard Deviation is equal to : ",s)

# results are as follow:

Xbar is equal to :  38.4675  

There are  800  number of rows in the data 

Standard Deviation is equal to :  2.753802

In [28]:
# In this section, I ran two methods to find Standard Error
# but  for reporting, I used method 2 to find Standard Error

# Method 1:
# Using qt function to calculate z-score

# z <- ((qt(0.995,df=(nrow(MyData))-1)))
# z = 2.58199654249689 <---- RESULT of qt function to find z score

# based on the qt function, I calculated Standard Error

# Standard Error using qt function in R <- ((qt(0.99,df=(nrow(MyData))-1))*s/(sqrt(nrow(MyData))))
# Standard Error = 0.226952200211327 <---- RESULT of  standard error by using qt function to find z score


# Method 2:

# I used the commonly used table which defines z equal to 2.576

se <- (2.576*s/(sqrt(nrow(MyData)))) # Z is equal 2.576 to Based on commonly used table
cat("Standard Error is equal to ---> ",se," \n\n")

cat("The lower CI of the sample is equal to ---> " ,(xmean - (2.576*s/(sqrt(nrow(MyData))))),  " \n\n") 
cat("The upper CI of the sample is equal to ---> ",(xmean + (2.576*s/(sqrt(nrow(MyData)))))," \n\n")

cat("In other words, 99% confidence interval is equal to  ---> (",(xmean - (2.576*s/(sqrt(nrow(MyData))))),",",(xmean + (2.576*s/(sqrt(nrow(MyData))))),") \n\n")

# Result of 99% confidence interval

Standard Error is equal to --->  0.2508035  

The lower CI of the sample is equal to --->  38.2167  

The upper CI of the sample is equal to --->  38.7183  

In other words, 99% confidence interval is equal to  ---> ( 38.2167 , 38.7183 ) 



## c) Conduct a hypothesis test evaluating whether the average weight gained is different from 30. Make a conclusion and explain your findings. 

$H0$: the average weight gained is not different from 30

$H1$: the average weight gained is  different from 30

In [29]:
w_gained <- MyData %>%
  select("gained")

# Testing the average weight gained by moms against hypothesized value
# hypothesized value >> mu0 = 30
t.test(w_gained$gained, mu = 30)
t.test(w_gained$gained, mu = 30, alternative = "less")
t.test(w_gained$gained, mu = 30, alternative = "greater")


	One Sample t-test

data:  w_gained$gained
t = 1.5538, df = 799, p-value = 0.1206
alternative hypothesis: true mean is not equal to 30
95 percent confidence interval:
 29.7966 31.7484
sample estimates:
mean of x 
  30.7725 



	One Sample t-test

data:  w_gained$gained
t = 1.5538, df = 799, p-value = 0.9397
alternative hypothesis: true mean is less than 30
95 percent confidence interval:
     -Inf 31.59121
sample estimates:
mean of x 
  30.7725 



	One Sample t-test

data:  w_gained$gained
t = 1.5538, df = 799, p-value = 0.06031
alternative hypothesis: true mean is greater than 30
95 percent confidence interval:
 29.95379      Inf
sample estimates:
mean of x 
  30.7725 


# Findings

To achieve a complete result, in testing if the avg gained weight is significantly different from 30, 

I tested with a normal t.test without any distribution transformations and ran the test three times:

### Hypothesis testing
1. if they are equal. 
Finding: p-value = 0.1206 (> .05) shows that they are not equal and as a result supporting the alternative hypothesis that the average weight gained is  different from 30.

### For further discoveries, I wanted to understand how different they are. So I tested:
2. if the average is less than 30.
Findings: p-value = 0.9397 (> .05) shows that average weight is not less than 30.

3. if the average is greater than 30.
Findings: p-value = 0.06031 ( slightly more than .05). So I check the sample estimates which indicates that mean of x = 30.7725

So I reject $H0$.